app/services/NotProcessedFilesService.scala

```diff
@@ -0,0 +1,105 @@
+import models.FailedJobResponses.FailedToProcessNotifications
+import models.MongoLockResponses
+import models.notification.RecordStatusEnum
+import play.api.Configuration
+import repositories.FileNotificationRepository
+import scheduler.ScheduleStatus.JobFailed
+import scheduler.{ScheduleStatus, ScheduledService}
+import uk.gov.hmrc.mongo.lock.{LockRepository, LockService, MongoLockRepository}
+import utils.Logger.logger
+import utils.PagerDutyHelper.PagerDutyKeys.{FAILED_TO_PROCESS_FILE_NOTIFICATION, MONGO_LOCK_UNKNOWN_EXCEPTION, NOTIFICATION_SET_TO_NOT_PROCESSED_PENDING_RETRY, UNKNOWN_EXCEPTION_FROM_SDES}
+import utils.{PagerDutyHelper, TimeMachine}
+
+import scala.concurrent.duration.{Duration, DurationInt}
+import scala.concurrent.{ExecutionContext, Future}
+
+class NotProcessedFilesService @Inject()(lockRepositoryProvider: MongoLockRepository,
+                                         fileNotificationRepository: FileNotificationRepository,
+                                         timeMachine: TimeMachine,
+                                         config: Configuration,
+                                         appConfig: AppConfig
+                                        )(implicit ec: ExecutionContext) extends ScheduledService[Either[ScheduleStatus.JobFailed, String]] {
+
+  val jobName = "NotProcessedFilesService"
+  lazy val mongoLockTimeoutSeconds: Int = config.get[Int](s"schedules.$jobName.mongoLockTimeout")
+
+  lazy val lockKeeper: LockService = new LockService {
+    override val lockId: String = s"schedules.$jobName"
+    override val ttl: Duration = mongoLockTimeoutSeconds.seconds
+    override val lockRepository: LockRepository = lockRepositoryProvider
+  }
+
+  //scalastyle:off
+  override def invoke: Future[Either[JobFailed, String]] = {
+    tryLock {
+      logger.info(s"[$jobName][invoke] - Job started")
+      for {
+        filesReceived <- fileNotificationRepository.getFilesReceivedBySDES()
+        filteredFiles = {
+          logger.info(s"[NotProcessedFilesService][invoke] - Number of files received by SDES: ${filesReceived.size}")
+          filesReceived.filter(notification => {
+            notification.nextAttemptAt.plusMinutes(appConfig.configurableTimeMinutes).isBefore(timeMachine.now)
+          })
+        }
+        sequenceOfResults <- Future.sequence(filteredFiles.map {
+          logger.info(s"[NotProcessedFilesService][invoke] - Number of filtered files: ${filteredFiles.size}")
+          wrapper => {
+            PagerDutyHelper.log("invoke", NOTIFICATION_SET_TO_NOT_PROCESSED_PENDING_RETRY)
+            logger.info(s"[NotProcessedFilesService][invoke] - Updating notification (reference: ${wrapper.reference}) to NOT_PROCESSED_PENDING_RETRY")
+            fileNotificationRepository.updateFileNotification(wrapper.reference, RecordStatusEnum.NOT_PROCESSED_PENDING_RETRY).map(_ => true)
+          }.recover {
+            case e => {
+              PagerDutyHelper.log("invoke", UNKNOWN_EXCEPTION_FROM_SDES)
```

Comment:
This PgaerDuty doesn't look right, we aren't receiving anything from SDES at this stage